### PR TITLE
[UI] Booking a user.

### LINKED ISSUE #35: [UI] Combining users and reservations.
Combining users and reservations.

### PR DESCRIPTION
**Pull request by Danil98Borisov**
Booking a user.

### CODE REVIEW

File: src/app/app.api.const.ts [hunk 1]
```diff
@@ -25,4 +25,6 @@ export class AppApiConst {
 
   public static RESERVATION_ALL = AppApiConst.API_ENDPOINT + '/reservation/all';
   public static RESERVATION_ADD = AppApiConst.API_ENDPOINT + '/reservation/add';
+
+  public static USER_BOOKING = AppApiConst.API_ENDPOINT + '/reservation/user/booking';
```
Comment (AndrewSmrnv) [inline src/app/app.api.const.ts]: why do we need /user/ here?

File: src/app/app.component.html [hunk 2]
```diff
@@ -7,6 +7,7 @@
     <a class="nav-item" routerLink="hotel/add">Add Hotel</a>
     <a class="nav-item" routerLink="hotel/edit">Edit Hotel</a>
     <a class="nav-item" routerLink="delete">Delete</a>
+    <a class="nav-item" routerLink="user/booking">My booking</a>
```
Comment (AndrewSmrnv) [inline src/app/app.component.html]: My bookingS ?

File: src/user-booking/user-booking.service.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export class UserBookingService {
+  constructor(private http: HttpClient) {
+  }
+
+  httpOptions = {
+    headers: new HttpHeaders({'Content-Type': 'application/json'})
+  };
+
+  getAllReservationUser(email: string): Observable<Reservation[]> {
+    const url = `${AppApiConst.USER_BOOKING}/${email}`;
```
Comment (AndrewSmrnv) [inline src/user-booking/user-booking.service.ts]: do we really need this const? why not to use it right in the eturn this.http.get<Reservation[]>(AppApiConst.USER_BOOKING + `/${email}` ?

File: src/user-booking/user-booking.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  ngOnInit() {
+
+    if (this.tokenStorage.getToken()) {
+      this.isLoggedIn = true;
+      this.roles = this.tokenStorage.getUser().roles;
+      const user = this.tokenStorage.getUser();
+      this.email = user.email;
+    }
```
Comment (AndrewSmrnv) [inline src/user-booking/user-booking.component.ts]: Why do you need this block? 
If email is the only necessary thing, pls use the UserService (add necessary method there)

File: src/models/bookingBy.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export class BookingBy {
```
Comment (AndrewSmrnv) [inline src/models/bookingBy.ts]: Is this object for Reservation request? If so, let's change the name to something more meaningful

File: src/app/app.api.const.ts (hunk 1 above)
Comment (Danil98Borisov) [reply src/app/app.api.const.ts]: done

File: src/app/app.component.html (hunk 2 above)
Comment (Danil98Borisov) [reply src/app/app.component.html]: done

File: src/user-booking/user-booking.service.ts (hunk 3 above)
Comment (Danil98Borisov) [reply src/user-booking/user-booking.service.ts]: done

**Pull request merged**